aten/src/ATen/cuda/CUDAMultiStreamGuard.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <c10/util/ArrayRef.h>
+#include <ATen/cuda/CUDAStream.h>
+#include <ATen/cuda/CUDAContext.h>
+
+#include <vector>
+
+namespace at { namespace cuda {
+
+// TODO: Implement this generically in c10.  You'll need some way to get
+// the number of GPUs from the GuardImpl, in that case.
+struct CUDAMultiStreamGuard {
```

Comment:
mark final

Reply:
Also, this is a class conceptually, not a struct, since it protects an invariant.